addons/mail/models/discuss/discuss_channel.py

```diff
@@ -316,9 +316,10 @@ def _subscribe_users_automatically(self):
     def _subscribe_users_automatically_get_members(self):
         """ Return new members per channel ID """
         return dict(
-            (channel.id, (channel.group_ids.users.partner_id - channel.channel_partner_ids).ids)
-            for channel in self
-        )
+            (channel.id,
+             ((channel.group_ids.users.partner_id.filtered(lambda u: u.active) - channel.channel_partner_ids).ids))
```

Comment:
Nitpick: `u` => `p` (because partner, not user)

Reply:
Thank you for your valuable suggestion, I have made the suggested changes.